R/utils.R

```diff
@@ -93,3 +93,22 @@ check_rel_susceptibility <- function(rel_susceptibility) {
       "increasing"))
   }
 }
+
+build_waning_rate <- function(waning_rate) {
+  if (length(waning_rate) == 0) {
+    stop("At least one value required for 'rel_susceptibility'")
+  }
+  N_age <- get_n_groups()
+  if (any(waning_rate < 0)) {
+    stop("'waning_rate' must have only non-negative values")
+  }
+  if (length(waning_rate) > 1) {
+    if (length(waning_rate) != N_age) {
+      stop("'waning_rate' should have as many elements as age groups")
+    }
+    vect_waning_rate <- waning_rate
+  } else { # create vector by repeating waning_rate for each age group
```

Comment:
This comment can be dropped unless you feel strongly. Alternatively, write this block as:

```r
if (length(waning_rate) == 1) {
  waning_rate <- rep(waning_rate, N_age)
} else if (length(waning_rate) != N_age) {
  stop("'waning_rate' should have as many elements as age groups, or be a scalar")
}
waning rate
```

(You do not need the `each`, taking the scalar branch first makes the logic clearer, and avoiding the rename means less is going on)